Gems/Atom/RPI/Code/Source/RPI.Edit/Material/MaterialSourceData.cpp

```diff
@@ -136,6 +136,19 @@ namespace AZ
                 return Failure();
             }
 
+            // Images are set to pre-load, so they will fully load when loading a material or material type asset.
+            // To create the material asset, we don't need to fully load the images that are referenced.
+            // So we use this filter to ignore the image assets
+            Data::AssetLoadParameters dontLoadImageAssets{ [](const AZ::Data::AssetFilterInfo& filterInfo)
+                                                           {
+                                                               if (filterInfo.m_assetType == AZ::AzTypeInfo<StreamingImageAsset>::Uuid())
```

Comment:
One thing I wasn't sure about here was if I need to check for other types like ImageAsset instead of StreamingImageAsset. StreamingImageAsset worked in practice.

Reply:
@VickyAtAZ recently added support for attachment images and possibly other types

Reply:
Those assets should be small, so for now I don't think there's a problem with checking just StreamingImageAsset. But if you were change this to check for all ImageAsset subclasses that would be more future proof.

Reply:
I've added AttachmentImageAsset and ImageAsset. Unfortunately there isn't a way that I can find to do this generically that would automatically handle future sub-classes of ImageAsset. If I had an actual live derived class object, I could use azrtti_istypeof<ImageAsset>(derivedObject). But I don't, I only have the TypeId of the derived object in filterInfo.m_assetType.